aggregator/aggregator.go

```diff
@@ -56,21 +61,35 @@ var AggregatorCommand = &cli.Command{
 			Value:   "http://localhost:4646",
 			Usage:   "HTTP API address of a Nomad server or agent.",
 		},
+		&cli.StringSliceFlag{
+			Name:    "dry-run-blacklist",
+			Aliases: []string{"d"},
+			Usage:   "Health checks in this blacklist, will not be dry-run.",
```

Comment:
Does this mean they won't be run at all?  "dry-run-blacklist" to me just means don't run the healthcheck at all.

Reply:
My bad, I should add some comments in the file, and may be update the `README.md` too.

Definition of dry-run, if the health check fails, don't take the node out of the scheduling pool, instead just do a dry-run i.e `print a message rather than performing the actual operation of setting SchedulingEligibility=false`

All health checks will be run (including the default ones CPUUnderPressure, MemoryUnderPressure and DiskUnderPressure) and by default all health checks will run in dry-run mode i.e if a health check fails on a particular node, node will **not** be taken out of the scheduling pool.

If operator sets a particular health check e.g. `docker` in the dry-run-blacklist : `--dry-run-blacklist=docker`.
If `docker` fails on a particular node, that node will become eligible to be toggled i.e that node could be taken out of the scheduling pool, provided if we take the node out, the number of eligible nodes are still within the threshold (default is 50%).

e.g. 
- if I am running my aggregator: `npd aggregator --dry-run-blacklist=docker --dry-run-blacklist portworx --threshold-percentage 40`
- `detector` is running with some custom health checks: `docker`, `portworx`, `ntp`
-  cluster has 100 nodes, 80/100 are eligible when we started the aggregator.

Let's say if on one of the nodes, `docker` starts failing. Since docker is in `dry-run-blacklist` this node can be taken out of the scheduling pool. Aggregator will then compute, if I take this node out, will I have sufficient capacity for future scheduling workloads. If the answer is `yes` based on `--threshold-percentage` it will take the node out, otherwise (in case of `no` it will just log a message and leave the node as-is).

Reply:
Based on Ted's feedback, we ll be changing `--dry-run-blacklist` to `--enforce-health-check`